core/src/main/java/com/datastrato/gravitino/authorization/UserGroupManager.java

```diff
@@ -58,21 +58,22 @@ public UserGroupManager(EntityStore store, IdGenerator idGenerator) {
    * @throws RuntimeException If adding the User encounters storage issues.
    */
   public User addUser(String metalake, String name) throws UserAlreadyExistsException {
-    UserEntity userEntity =
-        UserEntity.builder()
-            .withId(idGenerator.nextId())
-            .withName(name)
-            .withNamespace(
-                Namespace.of(
-                    metalake, Entity.SYSTEM_CATALOG_RESERVED_NAME, Entity.USER_SCHEMA_NAME))
-            .withRoles(Lists.newArrayList())
-            .withAuditInfo(
-                AuditInfo.builder()
-                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
-                    .withCreateTime(Instant.now())
-                    .build())
-            .build();
     try {
+      UserEntity userEntity =
+          UserEntity.builder()
+              .withId(idGenerator.nextId())
+              .withName(name)
+              .withNamespace(
+                  Namespace.of(
+                      metalake, Entity.SYSTEM_CATALOG_RESERVED_NAME, Entity.USER_SCHEMA_NAME))
+              .withRoles(Lists.newArrayList())
+              .withAuditInfo(
+                  AuditInfo.builder()
+                      .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                      .withCreateTime(Instant.now())
+                      .build())
+              .build();
+      AuthorizationUtils.checkMetalakeExists(store, metalake);
```

Comment:
Can we move this above, so that we don't have to execute the following codes if metalake is not existed?

Reply:
Done.